st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -267,18 +268,26 @@ class ActionExecutionReRunController(ActionExecutionsControllerMixin, ResourceCo
         'trigger_instance'
     ]
 
-    class ExecutionParametersAPI(object):
-        def __init__(self, parameters=None):
+    class ExecutionSpecificationAPI(object):
+        def __init__(self, parameters=None, tasks=None):
             self.parameters = parameters or {}
+            self.tasks = tasks or []
 
         def validate(self):
+            if self.tasks and self.parameters:
+                raise ValueError('Parameters override is not supported when '
+                                 're-running task(s) for a workflow.')
+
             if self.parameters:
                 assert isinstance(self.parameters, dict)
 
+            if self.tasks:
+                assert isinstance(self.tasks, list)
+
             return self
 
-    @jsexpose(body_cls=ExecutionParametersAPI, status_code=http_client.CREATED)
-    def post(self, execution_parameters, execution_id):
+    @jsexpose(body_cls=ExecutionSpecificationAPI, status_code=http_client.CREATED)
+    def post(self, spec, execution_id):
```

Comment:
Minor thing - for consistency it would be good to add `_api` suffix to the argument name (yeah, I know it wasn't not there earlier, but we have a chance to improve it now).